packages/mobile/src/identity/privateHashing.ts

```diff
@@ -71,6 +73,10 @@ function* doFetchPhoneHashPrivate(e164Number: string) {
   }
 
   Logger.debug(`${TAG}@fetchPrivatePhoneHash`, 'Salt was not cached, fetching')
+  const isBalanceSufficientForQuota = yield call(balanceSufficientForQuotaRetrieval)
+  if (!isBalanceSufficientForQuota) {
+    throw new Error(ErrorMessages.ODIS_INSUFFICIENT_BALANCE)
```

Comment:
This error message isn't going to be properly propagate because of the catch block of `fetchPhoneHashPrivate`. The way that catch block is currently setup it will only ever throw with a `SALT_QUOTA_EXCEEDED` or `SALT_FETCH_ERROR`. Probably easiest to add an `else if` clause to handle then propagate this specific error.